### ISSUE TITLE
Fix image number calculation for grid scans

### ISSUE THREAD

**Issue by DominicOram** [dials/nexgen#55]
If I have a gonio with axes like:
```    
"starts": [0.0],
"ends": [2.0],
"increments": [0.2],
```
then `calculate_grid_scan_range` calculates that there will be 10 images but actually I think there should be 11:
```
0.0, 0.2, 0.4, 0.6, 0.8, 1.0, 1.2, 1.4, 1.6, 1.8, 2.0
```
Subsequently when we pass this to `scanspec`, because we're assuming 10 images, it then gives us:
```
0.0, 0.222, 0.444, 0.666, 0.888 ... 2.0
```
Which disagrees with the increment number we've passed

**Comment by DominicOram** [comment]
There are also issues with floating point calculations. e.g.
```
    "starts": [420.9968],
    "ends": [1040.9968],
    "increments": [20],
```
gives a image length of 30

**Comment by noemifrisina** [comment]
In part fixed by #59 but it still needs a couple of fixes when using increments instead of number of images for a grid scan.
